metacat-connector-hive/src/main/java/com/netflix/metacat/hive/connector/ConnectorTableServiceImpl.java

```diff
@@ -0,0 +1,140 @@
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *     Unless required by applicable law or agreed to in writing, software
+ *     distributed under the License is distributed on an "AS IS" BASIS,
+ *     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *     See the License for the specific language governing permissions and
+ *     limitations under the License.
+ */
+
+package com.netflix.metacat.hive.connector;
+
+import com.google.inject.Inject;
+import com.netflix.metacat.common.MetacatRequestContext;
+import com.netflix.metacat.common.QualifiedName;
+import com.netflix.metacat.common.dto.Pageable;
+import com.netflix.metacat.common.dto.Sort;
+import com.netflix.metacat.common.dto.TableDto;
+import com.netflix.metacat.common.server.connectors.ConnectorTableService;
+import com.netflix.metacat.common.server.exception.TableAlreadyExistsException;
+import com.netflix.metacat.common.server.exception.TableNotFoundException;
+import com.netflix.metacat.hive.connector.converters.HiveMetacatConverters;
+import org.apache.hadoop.hive.metastore.api.Table;
+import org.apache.thrift.TException;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.List;
+
+/**
+ * Hive base connector base service impl.
+ *
+ * @author zhenl
+ */
+public class ConnectorTableServiceImpl implements ConnectorTableService {
+
+    private MetacatHiveClient metacatHiveClient;
+    private HiveMetacatConverters hiveMetacatConverters;
+
+    /**
+     * Constructor.
+     *
+     * @param metacatHiveClient     hiveclient
+     * @param hiveMetacatConverters converter
+     */
+    @Inject
+    public ConnectorTableServiceImpl(final MetacatHiveClient metacatHiveClient,
+                                     final HiveMetacatConverters hiveMetacatConverters) {
+        this.metacatHiveClient = metacatHiveClient;
+        this.hiveMetacatConverters = hiveMetacatConverters;
+    }
+
+    /**
+     * {@inheritdoc}.
+     */
+    @Override
+    public List<QualifiedName> listNames(
+        @Nonnull final MetacatRequestContext requestContext,
+        @Nonnull final QualifiedName name,
+        @Nullable final QualifiedName prefix,
+        @Nullable final Sort sort,
+        @Nullable final Pageable pageable
+    ) {
+        throw new UnsupportedOperationException(UNSUPPORTED_MESSAGE);
+    }
+
+    /**
+     * getTable.
+     *
+     * @param requestContext The request context
+     * @param name           The qualified name of the resource to get
+     * @return table dto
+     */
+    public TableDto get(@Nonnull final MetacatRequestContext requestContext, @Nonnull final QualifiedName name) {
+        try {
+            final Table table = metacatHiveClient.getTableByName(name.getDatabaseName(), name.getTableName());
+            return hiveMetacatConverters.hiveToMetacatTable(name, table);
+        } catch (TException exception) {
+            throw new TableNotFoundException(name);
```

Comment:
At least add the cause to the exception so we don't lose the stack trace and other info.